### PR TITLE
iiod: fix service start

### PR DESCRIPTION
**Pull request by cristina-suteu**
## PR Description
make sure iiod service is enabled even if only hwmon devices are present. fixes issue https://github.com/analogdevicesinc/libiio/issues/1194
## PR Type
- [x] Bug fix (a change that fixes an issue)
- [ ] New feature (a change that adds new functionality)
- [ ] Breaking change (a change that affects other repos or cause CIs to fail)

## PR Checklist
- [ ] I have conducted a self-review of my own code changes
- [ ] I have commented new code, particulary complex or unclear areas
- [ ] I have checked that I did not intoduced new warnings or errors (CI output)
- [ ] I have checked that components that use libiio did not get broken
- [ ] I have updated the documentation accordingly (GitHub Pages, READMEs, etc)

### CODE REVIEW

File: iiod/CMakeLists.txt [hunk 1]
```diff
@@ -90,8 +90,22 @@ if(NOT SKIP_INSTALL_ALL)
...
+		set(CMAKE_SERVICE_PATH "ConditionPathExists=|/sys/bus/iio\nConditionPathExists=|/sys/class/hwmon")
+	else()
+		set(CMAKE_SERVICE_PATH "ConditionPathExists=/sys/bus/iio")
+	endif()
 	configure_file(${CMAKE_CURRENT_SOURCE_DIR}/init/iiod.service.cmakein ${PROJECT_BINARY_DIR}/init/iiod.service)
 	install(FILES ${PROJECT_BINARY_DIR}/init/iiod.service DESTINATION ${SYSTEMD_UNIT_INSTALL_DIR})
+	find_program(SYSTEMCTL
+		NAMES systemctl
+		DOC "Control the systemd system and service manager (systemctl)")
+	if(SYSTEMCTL)
```
Comment (julienmalik) [inline iiod/CMakeLists.txt:103]: The part inside this if() will fail when cross-compiling

Comment (nunojsa) [reply iiod/CMakeLists.txt:103]: Yeah, moreover the change is not related with the current patch. And I honestly don't think handling like the below belongs to the build system. We should either let it for users or if we really want for the service to be automatically enabled, do it at the packet manager level.

**Review by nunojsa — COMMENTED** [review]
The hwmon para LGTM

**Review by nunojsa — APPROVED** [review]

**Review by adisuciu — APPROVED** [review]

**Comment by rgetz** [comment]
you should not do this in the release branch, without doing the same in the main branch.

**Pull request merged**